### DISCUSSION TITLE
NetworkXError: '_key' is not a valid key

### DISCUSSION THREAD

**Discussion by s-celles** [networkx/networkx#7440] [Q&A]
Hello,

I tried to export a graph from ArangoDB to GML thanks to NetworkX and [networkx-adapter](https://github.com/arangoml/networkx-adapter/)

I did

```python
from dotenv import load_dotenv
from arango import ArangoClient
from adbnx_adapter import ADBNX_Adapter
load_dotenv()
client = ArangoClient(hosts="http://localhost:8529")
db = client.db("My-DB", username="root", password=os.getenv("ARANGO_ROOT_PASSWORD"))
adbnx_adapter = ADBNX_Adapter(db)
nx_g = adbnx_adapter.arangodb_graph_to_networkx("My-Graph")
nx.write_gml(nx_g, "gh.gml")
```

but it raises

    NetworkXError: '_key' is not a valid key

I guess GML export doesn't support leading "_" in attributes.

So I did.

```python
for node, attr_dict in nx_g.nodes(data=True):
    node_data = {key.strip("_"): value for key, value in attr_dict.items()}
    nx.set_node_attributes(nx_g, node_data, node)

for edge in nx_g.edges(data=True):
    source, target, attr_dict = edge
    edge_data = {key.strip("_"): value for key, value in attr_dict.items()}
    nx.set_edge_attributes(nx_g, edge_data, f"{source}-{target}")

nx.write_gml(nx_g, "gh.gml")
```

but it raises

    ValueError: not enough values to unpack (expected 3, got 2)

This is probably quite similar to #7439

Kind regards

**Comment by Schefflera-Arboricola** [comment]
I don't think you can use `set_node/edge_attributes` to rename a node/edge attribute like this. Ref. docs :
- https://networkx.org/documentation/stable/reference/generated/networkx.classes.function.set_node_attributes.html
- https://networkx.org/documentation/stable/reference/generated/networkx.classes.function.set_edge_attributes.html#set-edge-attributes 

Thank you :)
